src/k8s/pkg/k8sd/types/cluster_config.go

```diff
@@ -43,9 +43,10 @@ type Certificates struct {
 }
 
 type Kubelet struct {
-	CloudProvider string `yaml:"cloud-provider,omitempty"`
-	ClusterDNS    string `yaml:"cluster-dns,omitempty"`
-	ClusterDomain string `yaml:"cluster-domain,omitempty"`
+	CloudProvider      string   `yaml:"cloud-provider,omitempty"`
+	ClusterDNS         string   `yaml:"cluster-dns,omitempty"`
+	ClusterDomain      string   `yaml:"cluster-domain,omitempty"`
+	RegisterWithTaints []string `yaml:"register-with-taints,omitempty"`
```

Comment:
Should the name state that these are only for control-plane nodes? e.g. `ControlPlaneTaints`?